### ISSUE TITLE
Compiling Fails w/o Compatibility mode in v7

### ISSUE THREAD

**Issue by kewalsky** [MScholtes/PS2EXE#14]
v7

```
PS C:\test> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.0.2
PSEdition                      Core
GitCommitId                    7.0.2
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS C:\test> Import-Module ps2exe
PS C:\test> Invoke-ps2exe -inputfile "EXE Test 01.ps1" -iconfile "EXE Test 01.ico"
PS2EXE-GUI v0.5.0.21 by Ingo Karstein, reworked and GUI support by Markus Scholtes


Reading input file C:\test\EXE Test 01.ps1
Compiling file...

MethodInvocationException: C:\Users\user\Documents\PowerShell\Modules\ps2exe\1.0.5\ps2exe.ps1:2575
Line |
2575 |      $cr = $cop.CompileAssemblyFromSource($cp, $programFrame)
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "CompileAssemblyFromSource" with "2" argument(s): "Operation is not supported on this platform."
```

v5
```
PS C:\test> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.19041.1
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.1
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1


PS C:\test> Import-Module ps2exe
PS C:\test> Invoke-ps2exe -inputfile "EXE Test 01.ps1" -iconfile "EXE Test 01.ico"
PS2EXE-GUI v0.5.0.21 by Ingo Karstein, reworked and GUI support by Markus Scholtes


Reading input file C:\test\EXE Test 01.ps1
Compiling file...

Output file C:\test\EXE Test 01.exe written
```

v7 w/ Compatibility
```
PS C:\test> $PSVersionTable                                                                                             
Name                           Value
----                           -----
PSVersion                      7.0.2
PSEdition                      Core
GitCommitId                    7.0.2
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS C:\test> Import-Module ps2exe -UseWindowsPowerShell
WARNING: The names of some imported commands from the module 'ps2exe' include unapproved verbs that might make them less discoverable. To find the commands with unapproved verbs, run the Import-Module command again with the Verbose parameter. For a list of approved verbs, type Get-Verb.
WARNING: Module ps2exe is loaded in Windows PowerShell using WinPSCompatSession remoting session; please note that all input and output of commands from this module will be deserialized objects. If you want to load this module into PowerShell Core please use 'Import-Module -SkipEditionCheck' syntax.
PS C:\test> Invoke-ps2exe -inputfile "EXE Test 01.ps1" -iconfile "EXE Test 01.ico"
PS2EXE-GUI v0.5.0.21 by Ingo Karstein, reworked and GUI support by Markus Scholtes


Reading input file C:\test\EXE Test 01.ps1
Compiling file...

Output file C:\test\EXE Test 01.exe written
```

**Comment by MScholtes** [comment]
Hello kewalsky,

the issue with PS2EXE and PowerShell 6 or 7 is the underlying .Net Core. PS2EXE compiles a frame C# program containing the PowerShell script. .Net Core (and so PowerShell 7) does not deliver a compiler anymore in its base installation, one has to install the SDK to get one ("Roslyn").

And once there is the compiler, the calling method has changed completely. I do not know how to implement this without weeks of work. If someone has a good idea how to make it "cheap", please tell.

Sorry, at the moment I see no possibility to provide support for Powershell Core.

Greetings

Markus

**Comment by kewalsky** [comment]
I buried it in there as the third example but it still does appear to run on Core v7 as long as you do "Import-Module ps2exe -UseWindowsPowerShell" before the Invoke-ps2exe command. I don't know if it would be feasible to have it detect the version of PowerShell it is running on and throw a warning or maybe just add an aside under the usage section of the Readme?

**Comment by MScholtes** [comment]
I will investigate what the best to do (but a message "Operation is not supported on this platform" is self speaking).

But first I go on vacation, more in three weeks...

**Comment by MScholtes** [comment]
Another option would be to download the script based version of **PS2EXE** from [TechNet-Gallery](https://github.com/MScholtes/TechNet-Gallery) or [PS2EXE-GUI: "Convert" PowerShell Scripts to EXE Files with GUI](https://gallery.technet.microsoft.com/PS2EXE-GUI-Convert-e7cb69d5) and compile it with itself.
You can then use the executable **ps2exe.exe** to compile from **Powershell Core** (unfortunately then the automatic parameter expansion is missing).

Greetings

Markus

**Comment by MScholtes** [comment]
Now back from vacation I will look for a solution or remedy

Greetings

Markus

**Comment by MScholtes** [comment]
Hello Keith,

I released the new version 1.0.6 of **PS2EXE** today with (limited) support for _PowerShell Core_. Since _PowerShell Core_ has no compiler built in, **PS2EXE** starts a new instance of _Windows PowerShell_ and runs **PS2EXE** there.

Hope this helps and works for you

Greetings

Markus

**Comment by kewalsky** [comment]
Yes I can confirm it compiled, thank you!
